include/ozo/result.h

```diff
@@ -406,7 +433,18 @@ class basic_result {
  *
  * @ingroup group-requests-types
  */
-using result = basic_result<ozo::pg::result>;
+class result final : public basic_result<pg::result> {
+public:
+    using basic_result<pg::result>::basic_result;
+    /**
+     * Transfers the underlying handle, if any, to a `ozo::shared_result` object. Multiple
+     * `ozo::shared_result` objects may reference the result handle, which is not possible
+     * with `ozo::result`. After calling `share()` on a `ozo::result`, `valid() == false`.
+     *
+     * @return shared_result_type - shared result object
+     */
+    shared_result share() { return pg::share_handle(std::move(handle()));}
```

Comment:
Intuitively, a method named like this does not take ownership of underlying value. How about to use something like `release_shared`?

Reply:
Well, I did it in a similar way as [std::future::share](https://en.cppreference.com/w/cpp/thread/future/share) just to conform to standard things with similar semantic.